game/game.gd

```diff
@@ -3,79 +3,116 @@ extends Node2D
 var port = 8877
 var ip = 'localhost'
 var max_players = 200
+var clients = []
+export var default_level: String = "res://level/level1.tscn"
 
 func _ready():
 	var peer = NetworkedMultiplayerENet.new()
 	var is_client = "--client" in OS.get_cmdline_args()
 	var is_dedicated = "--dedicated" in OS.get_cmdline_args()
 	
+	# Other arguments are in the style of "--arg=value"
+	for argument in OS.get_cmdline_args():
+		if argument.find("=") > -1:
+			var key_value = argument.split("=")
+			match key_value[0]:
+				'--ip': ip=key_value[1]
+				'--port': port=int(key_value[1])
+	
 	if is_client:
 		peer.create_client(ip, port)
-		get_tree().connect("server_disconnected", self, "client_note_disconnected")
+		assert(get_tree().connect("server_disconnected", self, "client_note_disconnected") == OK)
 	else:
 		print("Listening for connections on " + String(port) + " ...")
 		peer.create_server(port, max_players)
-		get_tree().connect("network_peer_connected", self, "server_player_connected")
-		get_tree().connect("network_peer_disconnected", self, "server_player_disconnected")
-		if not is_dedicated:
-			register_player(1, null, {})
+		assert(get_tree().connect("network_peer_connected", self, "server_client_connected") == OK)
+		assert(get_tree().connect("network_peer_disconnected", self, "server_client_disconnected") == OK)
 	
 	get_tree().set_network_peer(peer)
+	
+	# this must happen after the network peer is set
+	if not is_client:
+		if not is_dedicated:
+			register_client(1)
+		switch_level(default_level)
+
+func switch_level(level_path: String):
+	$level.set_level(level_path)
+	
+	for client in clients:
+		spawn_new_player(client.id)
+
+func spawn_new_player(id: int):
+	# inform all our players about the new player
+		
+		var new_player = spawn_object(String(id), $level.get_path(), "res://player/player.tscn", {})
+		new_player.id = id
+		spawn_object_on_clients(new_player)
+
+func spawn_object_on_clients(object: Node):
+	rpc("spawn_object", object.name, object.get_parent().get_path(), object.filename, object.get_node("sync").get_sync_state())
 
 func client_note_disconnected():
 	print("Server disconnected from player, exiting ...")
 	get_tree().quit()
 
-func server_player_connected(player_id: int):
-	if player_id != 1:
-		print("Connected ", player_id)
+func register_client(id: int):
+	var client = preload("res://game/client.gd").new()
+	client.id = id
+	clients.append(client)
+
+func remove_client(id: int):
+	var index = 0
+	for client in clients:
+		if client.id == id:
+			client.free()
+			clients.remove(index)
+			return
+		else:
+			index += 1
+
+func spawn_object_for(client_id: int, object: Node):
+	rpc_id(client_id, "spawn_object", object.name, object.get_parent().get_path(), object.filename, object.get_node("sync").get_sync_state())
```

Comment:
could be nice to assert that get_node("sync") is not null here and provide the user with a descriptive error (I suppose forgetting it can only happen when you create a custom player.tscn, so it's not that critical)

Reply:
You could also use spawn_object_on_clients/spawn_object_for to sync objects that have been spawned on the server, so you don't need a remotesync function to spawn the object, which might be complicated to construct.

I added an error check to both functions. If there is no 'sync' node, an error is printed, and no sync state is transferred, the object is still spawned though.